dnamic_analysis/output.py

```diff
@@ -0,0 +1,323 @@
+        row = 2
+        if len(sqlresults) > 0:
+            app_names = []
+            for app_name,username in sqlresults:
+                app_names.append(app_name)
+            for app_name in set(app_names):
+                self._excel_object.write(self._worksheet, self._col, row, app_name)
+                row += 1
+        self._col += 1
+        self._excel_object.save(self._workbook)
+
+
+    #################################################
+    ## Risky Expired Service Principal Names (SPN) ##
+    #################################################
+    def risky_spns(self,risky_spns, sqlcount):
+        # Write bulk data to Excel workbook
+        data = 'Total Unique Expired over Total Overall: {} / {}'.format(len(risky_spns), sqlcount)
+        self._excel_object.write(self._worksheet, self._col, 0, data, 'row1')
+        self._excel_object.write(self._worksheet, self._col, 1, 'Risky Expired Service Principal Names', 'header')
+        row = 2
+        for username,_ in risky_spns:
+            self._excel_object.write(self._worksheet, self._col, row, username)
+            row += 1
+        self._col += 1
+        self._excel_object.save(self._workbook)
+
+
+    #######################################
+    ## Hashes Found on Multiple Machines ##
+    #######################################
+    def hashes_found_on_multiple(self,uniquecount, admin_hash_found, totalsrv, totalwks, totaladminsrv, totaladminwks):
+        data = 'Total Unique Accounts: {}\n' \
+                'Total Administrative Hashes Found: {}\n' \
+                'Total on Workstations: {}\n' \
+                'Total on Servers: {}\n' \
+                'Total Admin Hashes on Workstations: {}\n' \
+                'Total Admin Hashes on Servers: {}'.format(uniquecount,len(admin_hash_found),totalwks,totalsrv,totaladminwks,totaladminsrv)
+        self._excel_object.write(self._worksheet, self._col, 0, data, 'row1')
+        self._excel_object.write(self._worksheet, self._col, 1, 'Hashes Found on Multiple Machines', 'header')
+        row = 2
+        for username in admin_hash_found:
+            self._excel_object.write(self._worksheet, self._col, row, username)
+            row += 1
+        self._col += 1
+        self._excel_object.save(self._workbook)
+
+
+    ##################################################################
+    ## Accounts Hashes Exposed on Multiple Machines - By %age Tiers ##
+    ##################################################################
+    def multi_machine_hashes(self,multi_machine_hashes):
+        tiers = ['> 95% Access','90-95% Access','80-90% Access','70-80% Access','60-70% Access',
+                '50-60% Access','40-50% Access','30-40% Access','20-30% Access','10-20% Access',
+                '< 10% Access']
+
+        self._excel_object.write(self._worksheet, self._col, 1, 'Accounts Hashes Exposed on Multiple Machines', 'header')
+
+        for i in range(len(multi_machine_hashes)):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/app/infamousjoeg/DNAmicAnalysis/pullRequest?prid=4572226)